state/tree.go

```diff
@@ -243,42 +243,47 @@ func (t *tree) debugPointer(ps []*hamt.Pointer) {
 	}
 }
 
+// GetAllActorsResult is the struct returned via a channel by the GetAllActors
+// method. This struct contains only an address string and the actor itself.
+type GetAllActorsResult struct {
+	Address string
+	Actor   *actor.Actor
+}
+
 // GetAllActors returns a slice of all actors in the StateTree, t.
-func GetAllActors(t Tree) ([]string, []*actor.Actor) {
+func GetAllActors(t Tree) <-chan GetAllActorsResult {
```

Comment:
I can't tell from the code why this has been changed to a channel. Is this because it's slow and so as to provide incremental results to the UI. Please document that if so (the comment above is now wrong in any case). 

(In future, it would be easier to review and discuss if the functional change were in a separate PR).

Reply:
This change was suggested by @acruikshank because GetAllActors could potentially return a very large number of actors. I've updated the comment.